app/script/conversation/ConversationRepository.js

```diff
@@ -3392,16 +3392,19 @@ z.conversation.ConversationRepository = class ConversationRepository {
 
   handleMessageExpiration(messageEntity) {
     amplify.publish(z.event.WebApp.CONVERSATION.EPHEMERAL_MESSAGE_TIMEOUT, messageEntity);
-    if (!messageEntity.user().is_me) {
-      this.get_conversation_by_id(messageEntity.conversation_id).then(conversationEntity => {
-        if (conversationEntity.is_group()) {
-          const user_ids = _.union([this.selfUser().id], [messageEntity.from]);
-          return this.delete_message_everyone(conversationEntity, messageEntity, user_ids);
-        }
-
-        return this.delete_message_everyone(conversationEntity, messageEntity);
-      });
+    const shouldDeleteMessage = !messageEntity.user().is_me || messageEntity.is_ping();
+    if (!shouldDeleteMessage) {
+      return;
     }
+
+    this.get_conversation_by_id(messageEntity.conversation_id).then(conversationEntity => {
+      if (conversationEntity.is_group()) {
+        const user_ids = _.union([this.selfUser().id], [messageEntity.from]);
+        return this.delete_message_everyone(conversationEntity, messageEntity, user_ids);
+      }
+
+      return this.delete_message_everyone(conversationEntity, messageEntity);
+    });
```

Comment:
This is so short. Let stick with the positive case and not do and early return. There isn't much indentation either.